### PR TITLE
Docs on configuring KSQL -> SR over HTTPS

### PR DESCRIPTION
**Pull request by big-andy-coates**
Doc part of #950

(Which got lost due to doc location differences)

### CODE REVIEW

File: docs-rst/installation/server-config/security.rst [hunk 1]
```diff
@@ -29,27 +30,46 @@ You can use KSQL with a Kafka cluster in |ccloud|. For more information, see :re
 Configuring KSQL for Secured Confluent Schema Registry
 ------------------------------------------------------
 
-The following configuration connects KSQL with the Confluent Schema Registry over HTTPS.
+KSQL can be configured to connect to the Schema Registry over HTTP by setting the
+``ksql.schema.registry.url`` to the Schema Registry's HTTPS endpoint.
+Depending on your security setup, you may also need to supply a trust-store, (e.g. if the Schema
```
Comment (joel-hamill) [inline docs-rst/installation/server-config/security.rst]: `Depending on your security setup, you might also need to supply a trustStore and a keyStore. For example, a trustStore is required if the Schema Registry's SSL certificates are not trusted by the JVM by default; a keyStore is required if the Schema Registry requires mutual authentication.`

File: docs-rst/installation/server-config/security.rst [hunk 2]
```diff
@@ -29,27 +30,46 @@ You can use KSQL with a Kafka cluster in |ccloud|. For more information, see :re
...
+Registry's SSL certificates are not trusted they the JVM by default), and a key-store, (e.g. if the
+Schema Registry requires mutual authentication).
 
-#. Specify the HTTPS endpoint in the ``ksql.schema.registry.url`` setting in the
-   KSQL server configuration file:
+The following configuration is all that is needed to configure KSQL to communicate with the Schema Registry
```
Comment (joel-hamill) [inline docs-rst/installation/server-config/security.rst]: Simplify as `To configure KSQL communication with the Schema Registry over HTTPS, where mutual authentication is not required and the Schema Registry's SSL certificates are trusted by the JVM:`

File: docs-rst/installation/server-config/security.rst [hunk 3]
```diff
@@ -29,27 +30,46 @@ You can use KSQL with a Kafka cluster in |ccloud|. For more information, see :re
...
+over HTTPS, where mutual authentication is not required and the Schema Registry's SSL certificates
+are trusted by the JVM:
 
-   .. code:: bash
+.. code:: bash
+
+    ksql.schema.registry.url=https://<host-name-of-schema-registry>:<ssl-port>
+
+The following configuration can be used to configure KSQL to communicate with the Schema Registry over HTTPS,
```
Comment (joel-hamill) [inline docs-rst/installation/server-config/security.rst]: Simplify as: `To configure KSQL to communicate with the Schema Registry over HTTPS, with mutual authentication (i.e. SSL configuration is shared between Kafka and Schema Registry):`

File: docs-rst/installation/server-config/security.rst [hunk 4]
```diff
@@ -29,27 +30,46 @@ You can use KSQL with a Kafka cluster in |ccloud|. For more information, see :re
...
+    ssl.truststore.location=/etc/kafka/secrets/ksql.truststore.jks
+    ssl.truststore.password=confluent
+    ssl.keystore.location=/etc/kafka/secrets/ksql.keystore.jks
+    ssl.keystore.password=confluent
+    ssl.key.password=confluent
 
-   For example, if the Schema Registry's SSL certificate is not signed by a CA that is recognized by
-   the JVM by default, then you can provide a suitable truststore when starting KSQL via the command line:
+If required, specific configuration for the Schema Registry can be supplied. The following configuration
+can be used to configure KSQL to communicate with the Schema Registry over HTTP, _without_ mutual auth,
```
Comment (joel-hamill) [inline docs-rst/installation/server-config/security.rst]: Simplify (and remove unnecessary formatting): `Use following to configure KSQL to communicate with the Schema Registry over HTTP without mutual authentication (i.e. SSL configuration is not shared):`

**Comment by big-andy-coates** [comment]
retest this please

File: docs-rst/installation/server-config/security.rst (hunk 1 above)
Comment (big-andy-coates) [reply docs-rst/installation/server-config/security.rst]: done

File: docs-rst/installation/server-config/security.rst (hunk 3 above)
Comment (big-andy-coates) [reply docs-rst/installation/server-config/security.rst]: This reworking doesn't work: 'mutual authentication' is not the same thing as sharing config between Kafka and Schema Registry.  

Mutual authentication is a for of SSL communication where both parties authenticate with each other using SSL certificates. 

Shared configuration relates to the fact that you can configure SSL in the SR client using either prefixed or non prefixed configuration names.  e.g. you can use either `ssl.truststore.location` or `ksql.schema.registry.ssl.truststore.location`.  The former is used where the same SSL configuration is needed for communicating with both Kafka and the SR. The latter overrides the non-prefixed and only configures SR SSL comms.

I've added a part to explain this at the top of the section and reworked this paragraph. See what you think.

File: docs-rst/installation/server-config/security.rst (hunk 4 above)
Comment (big-andy-coates) [reply docs-rst/installation/server-config/security.rst]: As above, this reworking doesn't work. I've changed things around in my next commit. So see what you think.
